src/main/java/com/onarandombox/MultiversePortals/PortalPlayerSession.java

```diff
@@ -282,31 +282,60 @@ public void setTeleportTime(Date date) {
     }
 
-    public boolean allowTeleportViaCooldown(Date date) {
-        return (date.after(new Date(this.lastTeleportTime.getTime() + this.plugin.getCooldownTime())));
+    /**
+     * Get the remaining teleport cooldown time in milliseconds.
+     * If the value returned is not positive, the cooldown is no longer in effect.
+     *
+     * @return The remaining teleport cooldown time in milliseconds. Note that a
+     *         negative value may be returned if the cooldown is no longer in effect.
+     */
+    private long getRemainingTeleportCooldown() {
+        long cooldownEndMs = this.lastTeleportTime.getTime() + this.plugin.getCooldownTime();
+        long timeMs = (new Date()).getTime();
+        return cooldownEndMs - timeMs;
     }
 
-    public String getFriendlyRemainingTimeMessage() {
-        String remaining = "There is a portal " + ChatColor.AQUA + "cooldown" + ChatColor.WHITE + " in effect. Please try again in " + ChatColor.GOLD
-        + Integer.toString((int) this.getRemainingCooldown() / 1000) + "s.";
-        int time = (int) this.getRemainingCooldown() / 1000;
-        // Account for the fact that 0 seconds means 1
-        time++;
-        if (time <= 0) {
-            remaining += "< 1s" + ChatColor.WHITE + ".";
-        } else {
-            remaining += time + "s" + ChatColor.WHITE + ".";
+    /**
+     * Converts a long representing a time in milliseconds to a human-readable String.
+     *
+     * @param cooldownMs Time in milliseconds.
+     * @return Human-readable String with the given time in seconds.
+     */
+    private String formatCooldownTime(long cooldownMs) {
+        if (cooldownMs < 1000) {
+            return "1s";
         }
-        return remaining;
+
+        return (cooldownMs / 1000) + "s";
     }
 
-    public long getRemainingCooldown() {
-        //Calculate the remaining cooldown period
-        long remainingcooldown = (this.plugin.getCooldownTime() - ((new Date()).getTime() - this.lastTeleportTime.getTime()));
-        if (remainingcooldown < 0) {
-            remainingcooldown = 0;
+    /**
+     * Constructs a message informing a player about the
+     * remaining cooldown time.
+     *
+     * @param cooldownMs The cooldown time in milliseconds.
+     * @return           A message to be sent to a player, informing them about the remaining cooldown time.
+     */
+    private String getCooldownMessage(long cooldownMs) {
+        return "There is a portal " + ChatColor.AQUA + "cooldown "
+                + ChatColor.WHITE + "in effect. Please try again in "
+                + ChatColor.GOLD + this.formatCooldownTime(cooldownMs)
+                + ChatColor.WHITE + ".";
+    }
+
+    /**
+     * Checks if the teleport cooldown is still in effect. If it is, a message is sent
+     * to the player informing them.
+     *
+     * @return True if the teleport cooldown is still in effect, false otherwise.
+     */
+    public boolean checkAndSendCooldownMessage() {
```

Comment:
Usually public classes are above the private class?

Reply:
I've never heard of that convention, but sure, I'll change it.

Reply:
Alright, should be good to go now.